src/Middleware/RequestThrottling/src/Internal/LoggerExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.RequestThrottling.Internal
+{
+    internal static class LoggerExtensions
+    {
+        private static readonly Action<ILogger, Exception> _requestEnqueued;
+        private static readonly Action<ILogger, Exception> _requestDequeued;
+
+        static LoggerExtensions()
+        {
+            _requestEnqueued = LoggerMessage.Define(
+                logLevel: LogLevel.Debug,
+                eventId: new EventId(1, "Request Enqueued"),
+                formatString: "Server is busy; queuing request in middleware."
+                );
+            _requestDequeued = LoggerMessage.Define(
+                logLevel: LogLevel.Debug,
+                eventId: new EventId(2, "Request Dequeued"),
+                formatString: "Space availible on server; request is leaving queue."
+                );
+        }
+
+        internal static void RequestEnqueued(this ILogger logger)
```

Comment:
Internal or not, I think it's a bad idea to make log methods extension methods because it makes it too easy to log messages to the wrong logging namespace.

I would either just make these normal static methods like [DefaultHubDispatcher.Log.cs](https://github.com/aspnet/AspNetCore/blob/01a87989364dea1b18743a9fe3757eccc220b2ca/src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.Log.cs) or create a logger object that has a private field reference the correct logger like [KestrelTace.cs](https://github.com/aspnet/AspNetCore/blob/01a87989364dea1b18743a9fe3757eccc220b2ca/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs).

Reply:
@halter73? Extension methods are the standard pattern we use.
https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/StaticFiles/src/LoggerExtensions.cs
https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Formatters.Xml/src/LoggerExtensions.cs
https://github.com/aspnet/AspNetCore/blob/master/src/Servers/Kestrel/Core/src/Internal/LoggerExtensions.cs

Reply:
Gross. Even in Kestrel.Core! 😭 I think I should have made you put those methods in KestrelTrace.

Clearly we're inconsistent with whether we use extension methods on `ILogger` for structured logging or not. I vote for not using extension methods for this purpose going forward.

On an unrelated note, it looks like the consumers of Kestrel's LoggerExtensions are logging to `Microsoft.AspNetCore.Server.Kestrel.Core.KestrelServer` instead of `Microsoft.AspNetCore.Server.Kestrel` like the rest of the Kestrel.Core assembly. I guess at least we don't have EventId conflicts that way... Do you think it's worth consolidating these namespaces before 3.0, or is it not worth the break?

Reply:
We changed the pattern but didn’t update all the places, nor was it really communicated broadly on the team.

Reply:
> We changed the pattern but didn’t update all the places, nor was it really communicated broadly on the team.

To the extension method pattern, right?

Reply:
Nope. Static method

Reply:
@davidfowl MAKE UP YOUR MINDS. Put it in https://github.com/aspnet/AspNetCore/wiki/Engineering-guidelines or never speak of this again.

Reply:
@rynowak do the needful

Reply:
https://github.com/aspnet/AspNetCore/wiki/Engineering-guidelines#logging-patterns